packages/webpack-config-single-spa/package.json

```diff
@@ -23,6 +23,7 @@
   },
   "dependencies": {
     "clean-webpack-plugin": "^3.0.0",
+    "fork-ts-checker-webpack-plugin": "^4.1.3",
```

Comment:
This is used here: https://github.com/single-spa/create-single-spa/blob/master/packages/webpack-config-single-spa/lib/webpack-config-single-spa.js#L5

But it's not in the package.json so it isn't bundled into the npm module and the npm module cannot run without it.

Reply:
If we add this, we also will need to add `typescript` as a dependency.

Hmm I'd hate to force everyone to install fork-ts-checker-webpack-plugin and typescript even when they're not using them. One option could be to mark them as optionalDependencies?? To hint at the need for them to be installed. Then we'd update the webpack config to only call `require("fork-ts-checker-webpack-plugin")` inside of the ternary so that it's avoided when typescript is not needed?

Alternatively we just bite the bullet and add both of them as dependencies. Or even split this package into two, so that we can have a separate package.json for typescript dependencies. Thoughts?

Reply:
I honestly think webpack-config-single-spa-ts might be best?

Reply:
This is probably naive but could we just do a dynamic require here instead?

https://github.com/single-spa/create-single-spa/blob/master/packages/webpack-config-single-spa/lib/webpack-config-single-spa.js#L99

I'm not sure I understand why typescript is a dependency of that plugin

Reply:
Or could we have a `webpack-config-single-spa-ts` that just uses webpack merge to add those options to `webpack-config-single-spa`?

Reply:
Yes that's what I had in mind. webpack-config-single-spa-ts would use webpack-config-single-spa.

Reply:
> This is probably naive but could we just do a dynamic require here instead?

Yes that is exactly what I suggested in my comment:

> Then we'd update the webpack config to only call require("fork-ts-checker-webpack-plugin") inside of the ternary so that it's avoided when typescript is not needed

Reply:
> Yes that is exactly what I suggested in my comment
🤦‍♂  I read that wrong. Whoops.

As far as dynamic require vs `webpack-config-single-spa-ts` I'm good with either option. What's your preference? This MR won't solve it so I'll close it.